src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+using Microsoft.VisualStudio.LanguageServices.FindUsages;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SymbolSearch
+{
+    internal class RoslynSymbolSearchResult : SymbolSearchResult, IResultWithFileLocation, IResultWithContext, IResultWithNamedProject, IResultWithVSGuids, IResultWithReferenceOrDefinition, IResultWithIcon, IResultWithCustomData /* takes care of containin member, containing type and kind */
+    {
+        private DefinitionItem DefinitionItem { get; set; }
+        private SourceReferenceItem ReferenceItem { get; set; }
+        public SymbolSearchContext Context { get; }
+        private SymbolSearchSource SourceInternal { get; }
+        private string PlainText { get; set; }
+
+        private RoslynSymbolSearchResult(SymbolSearchContext context, string plainText) : base(context.SymbolSource, context.LocalOrigin, plainText)
+        {
+            this.Context = context;
+            this.SourceInternal = context.SymbolSource;
+        }
+
+        internal static async Task<RoslynSymbolSearchResult> MakeAsync(SymbolSearchContext context, DefinitionItem definition, DocumentSpan documentSpan, CancellationToken token)
+        {
+            return await MakeResultAsync(context, documentSpan, referenceItem: null, definitionItem: definition, cancellationToken: token).ConfigureAwait(false);
+        }
+
+        internal static async Task<RoslynSymbolSearchResult> MakeAsync(SymbolSearchContext context, SourceReferenceItem reference, DocumentSpan documentSpan, CancellationToken token)
+        {
+            return await MakeResultAsync(context, documentSpan, referenceItem: reference, definitionItem: reference.Definition, cancellationToken: token).ConfigureAwait(false);
+        }
+
+        private static async Task<RoslynSymbolSearchResult> MakeResultAsync(SymbolSearchContext context, DocumentSpan documentSpan, SourceReferenceItem referenceItem, DefinitionItem definitionItem, CancellationToken cancellationToken)
+        {
+            var (projectGuid, documentGuid, projectName, sourceText) = await FindUsagesUtilities.GetGuidAndProjectNameAndSourceTextAsync(documentSpan.Document, cancellationToken)
+                .ConfigureAwait(false);
```

Comment:
an you wrap these very long lines.  if they don't even show up on github, they're really too long to review effectively.